### PR TITLE
docker events and net-util

### PR DESCRIPTION
**Pull request by cjellick**

### CODE REVIEW

File: apply.sh [hunk 1]
```diff
@@ -8,7 +8,8 @@ cd $(dirname $0)
 
 mkdir -p ${CATTLE_HOME}/bin
 
-cp bin/host-api ${CATTLE_HOME}/bin
+cp bin/host-api bin/net-util.sh ${CATTLE_HOME}/bin
```
Comment (ibuildthecloud) [inline apply.sh]: In scripts/package copy from scripts/net-utils.sh to bin/net-utils.sh

File: events/event_router.go [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+	if err := e.dockerClient.RemoveEventListener(e.listener); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (e *EventRouter) routeEvents() {
+	for {
+		event := <-e.listener
+		w := <-e.workers
```
Comment (ibuildthecloud) [inline events/event_router.go]: Add timeout just so that we can get some logging, but don't ever drop event.

File: events/network/network.go [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+	log.Infof(string(output[:]))
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func buildCommand(pid string, ip string) *exec.Cmd {
+	if runtime.GOOS == "linux" {
```
Comment (ibuildthecloud) [inline events/network/network.go]: key off of env variable

File: events/network/network.go [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+	ConfigureIP(pid string, ip string) error
+}
+
+type NetUtilImpl struct{}
+
+func (n *NetUtilImpl) ConfigureIP(pid string, ip string) error {
+	command := buildCommand(pid, ip)
+
+	output, err := command.CombinedOutput()
+	log.Infof(string(output[:]))
```
Comment (ibuildthecloud) [inline events/network/network.go]: [:] not needed

File: events/start_handler.go [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+	Client StartHandlerDockerClient
+}
+
+type StartHandlerDockerClient interface {
+	InspectContainer(id string) (*docker.Container, error)
+}
+
+func (h *StartHandler) Handle(event *docker.APIEvents) error {
+	// Need to check that the container is running
+	lock := locks.Lock(event.ID)
```
Comment (ibuildthecloud) [inline events/start_handler.go]: add comment that event.ID is container ID

File: events/start_handler.go [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+	if lock == nil {
```
Comment (ibuildthecloud) [inline events/start_handler.go]: not blocking could cause an issue, but this seems like right behavior now

Comment (ibuildthecloud) [inline events/start_handler.go]: maybe lock on event type + container ID

File: events/start_handler.go [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+		return err
+	}
+
+	rancherIp := h.getRancherIp(c)
+	if rancherIp == "" {
+		return nil
+	}
+
+	if !c.State.Running {
+		log.Infof("Container [%s] not running. Cant assign IP [%s].", c.ID, rancherIp)
```
Comment (ibuildthecloud) [inline events/start_handler.go]: typo `Can't`

File: .gitignore [hunk 8]
```diff
@@ -1,4 +1,5 @@
-bin
+host-api
```
Comment (ibuildthecloud) [inline .gitignore]: Can we ignore bin again?

File: events/client.go [hunk 9]
```diff
@@ -9,10 +9,11 @@ import (
 
 const (
 	defaultUnixSocket = "unix:///var/run/docker.sock"
+	// TODO TO 1.17
```
Comment (ibuildthecloud) [inline events/client.go]: Why todo?  Do you need something from 1.17?  We are stuck at Docker 1.4 atm.

File: .gitignore (hunk 8 above)
Comment (cjellick) [inline .gitignore]: yes. will update.

File: events/client.go (hunk 9 above)
Comment (cjellick) [inline events/client.go]: no need. will remove todo

File: .gitignore (hunk 8 above)
Comment (cjellick) [inline .gitignore]: Ultimately, shoiuld pull this from an environment variable, like python-agent. I'd like to keep it like this for now and put in a separate issue for that.

**Comment by cjellick** [comment]
Ammended commit. Only changes were bin in .gitignore and removing TODO

**Comment by ibuildthecloud** [comment]
LGTM

**Pull request merged**